test/plugin_test.dart

```diff
@@ -14,50 +16,119 @@ void main() {
     injector.registerSingleton<ProcessManager>(() => pm, override: true);
     fs = MemoryFileSystem.test();
+    fs.currentDirectory = fs.directory(buildDir)..createSync();
     injector.registerSingleton<FileSystem>(() => fs, override: true);
+    injector.registerSingleton<CLISetup>(() => MockCLI(), override: true);
   });
 
-  test('fails without args and pubspec', () async {
-    final exitCode = await plugin.run([]);
-    expect(exitCode, 1);
-    expect(pm.commandLog, const [
-      'chmod +x .dart_tool/pub/bin/sentry_dart_plugin/sentry-cli',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli help'
-    ]);
-  });
+  Future<Iterable<String>> runWith(String config) async {
+    // properly indent the configuration for the `sentry` section in the yaml
+    final configIndented =
+        config.trim().split('\n').map((l) => '  ${l.trim()}').join('\n');
 
-  test('works with pubspec', () async {
     fs.file('pubspec.yaml').writeAsStringSync('''
-name: project
-version: 1.1.0
+name: $project
+version: $version
 
 sentry:
-  upload_native_symbols: true
-  include_native_sources: true
-  upload_source_maps: true
-  auth_token: t
+  auth_token: t # TODO: support not specifying this, let sentry-cli use the value it can find in its configs
   project: p
   org: o
-  url: http://127.0.0.1
-  log_level: debug
+$configIndented
 ''');
+
     final exitCode = await plugin.run([]);
     expect(exitCode, 0);
-    expect(pm.commandLog, const [
-      'chmod +x .dart_tool/pub/bin/sentry_dart_plugin/sentry-cli',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli help',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli --url http://127.0.0.1 --auth-token t --log-level debug upload-dif --include-sources --org o --project p /',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli --url http://127.0.0.1 --auth-token t --log-level debug releases --org o --project p new project@1.1.0',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli --url http://127.0.0.1 --auth-token t --log-level debug releases --org o --project p files project@1.1.0 upload-sourcemaps /build/web --ext map --ext js',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli --url http://127.0.0.1 --auth-token t --log-level debug releases --org o --project p files project@1.1.0 upload-sourcemaps / --ext dart',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli --url http://127.0.0.1 --auth-token t --log-level debug releases --org o --project p finalize project@1.1.0'
+    expect(pm.commandLog.take(2), const ['chmod +x $cli', '$cli help']);
+    return pm.commandLog.skip(2);
+  }
+
+  test('fails without args and pubspec', () async {
+    final exitCode = await plugin.run([]);
+    expect(exitCode, 1);
+    expect(pm.commandLog, const ['chmod +x $cli', '$cli help']);
+  });
+
+  test('works with pubspec', () async {
+    // TODO: because `url` param affects all commands, make it a test-group argument and run all test cases with/without it.
```

Comment:
Something that we should address in this PR?

Reply:
didn't want to blow this up too much but since you've already seen tests without that change, I've done it now.